src/pyfmi/fmi_util.pyx

```diff
@@ -151,20 +154,19 @@ cpdef prepare_data_info(np.ndarray[int, ndim=2] data_info, list sorted_vars, lis
             data_info[0,i] = last_data_matrix
             data_info[1,i] = alias*last_index
         else:
-            variability = var.variability
             last_vref   = var.value_reference
-            data_type   = var.type
```

Comment:
Not sure if this change improves things? It seems sensible that these computations are broken out?

Reply:
I agree that it is more readable, but not sure to which degree the compilation of Python code can optimize this. Namely the question that these class attributes do not have a constant variability, yet are not explicitly changed in between.

(Feel free to decide & resolve)

Reply:
Hmm I'll leave this as it is now and I think when we do a wider performance investigation it will show up if significant. (I think we should benchmark and compare with detailed output between FMI2 and FMI3 in a later stage.